### PR TITLE
Feature/user messages

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do (summary of changes)?
 - Adds user message for search results
 - Adds user message for status code errors
 
## How should it be tested?
 - Make a valid and invalid search to see search messages
 - Stop server to check status code error message
 
## Future Iterations:
 - We can mix up the language or styling of the messages if you want!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -74,11 +84,13 @@ class App extends Component {
...
           <MySongLibrary
-          songs={ this.state.mySongs } 
+          songs={ this.state.mySongs }
           mySongs={this.state.mySongs }
-          handleSong={this.removeSong} 
+          handleSong={this.removeSong}
           buttonIcon={[<MdRemoveCircle className="handle-song-icon"/>]}
           />}
+          {this.state.error &&
+          <h2>{this.state.error}</h2>}
```
Comment (monshan) [inline src/App.js:93]: we can add a class here for styling, but otherwise nice work on the conditional render

File: src/App.js [hunk 2]
```diff
@@ -90,6 +102,8 @@ class App extends Component {
           handleSong={this.addSong} 
           buttonIcon={[<RiHeartAddLine className="handle-song-icon"/> , <FaHeart className="heart-icon" data-cy="heart-icon"/>]}
           />}
+          {this.state.error &&
+          <h2>{this.state.error}</h2>}
```
Comment (monshan) [inline src/App.js:106]: Might be small considering its just twice but maybe put this into a helper function?

File: src/components/SongLibrary/SongLibrary.js [hunk 3]
```diff
@@ -8,33 +8,40 @@ import '../SongLibrary/SongLibrary.scss';
...
     || song.artist.toUpperCase().includes(modifiedSearchQuery)
     )
-    this.setState({ renderedSongs: matchingSongs })
+
+    if(matchingSongs !== "") {
+      this.setState({ searchResultsMsg: `Showing results for '${searchQuery.toLowerCase()}'':` })
+      this.setState({ renderedSongs: matchingSongs })
+    } else {
+      this.setState({ searchResultsMsg: "No results for this search. Time to freestyle! (Or try another search 😉)" })
+      this.setState({ renderedSongs: [] })
```
Comment (monshan) [inline src/components/SongLibrary/SongLibrary.js:29]: ooooh nice didn't think about that

**Review by monshan — APPROVED** [review]

File: src/components/SongLibrary/SongLibrary.js (hunk 3 above)
Comment (elizhahn) [inline src/components/SongLibrary/SongLibrary.js:29]: we could combine these into one this.setState()

Comment (elizhahn) [reply src/components/SongLibrary/SongLibrary.js:29]: this.setState({ searchResultsMsg: `Showing results for '${searchQuery.toLowerCase()}'':` , renderedSongs: []})

**Review by elizhahn — APPROVED** [review]

**Comment by m1073496** [comment]
[Add message for no search results](https://trello.com/c/opoQERqE/77-add-message-for-no-search-results)

**Comment by m1073496** [comment]
[Add helpful status code errors for users](https://trello.com/c/N7dHKOYo/78-add-helpful-status-code-errors-for-users)

**Pull request merged**